crates/searchers/openapi.yml

```diff
@@ -0,0 +1,642 @@
+openapi: 3.0.3
+info:
+  title: Searcher API
+  description: |
+    The API implemented by searchers interacting with the reference driver
+    implementation.
+  version: 0.0.1
+
+paths:
+  /:
+    post:
+      description: |
+        Solve the passed in auction instance.
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/Auction"
+      responses:
+        200:
+          description: Auction successfully solved.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Solution"
+        400:
+          description: There is something wrong with the requests.
+        500:
+          description: Something went wrong when handling the request.
+
+components:
+  schemas:
+
+    Address:
+      description: |
+        An Ethereum public address.
+      type: string
+      example: "0x0000000000000000000000000000000000000000"
+
+    Token:
+      description: |
+        An ERC20 token address.
+      type: string
+      example: "0xDEf1CA1fb7FBcDC777520aa7f396b4E015F497aB"
+
+    TokenAmount:
+      description: |
+        Amount of an ERC20 token. 256 bit unsigned integer in decimal notation.
+      type: string
+      example: "1234567890"
+
+    BigInt:
+      description: |
+        An arbitrary-precision integer value.
+      type: string
+      example: "1234567890"
+
+    Decimal:
+      description: |
+        An arbitrary-precision decimal value.
+      type: string
+      example: "13.37"
+
+    Digest:
+      description: |
+        A hex-encoded 32 byte digest.
+      type: string
+      example: "0x1e66721bb1bd77d2641c77ea1d61e8abb92bf69c64fcc90c2c6ad518d1b50db1"
+
+    NativePrice:
+      description: |
+        The price in wei of the native token (Ether on Mainnet for example) to
+        buy 10**18 of a token.
+      type: string
+      example: "1234567890"
+
+    TokenInfo:
+      description: |
+        Information about a token relavant to the auction.
+      type: object
+      properties:
+        decimals:
+          type: integer
+          description: |
+            The ERC20.decimals value for this token.
+        symbol:
+          type: string
+          description: |
+            The ERC20.symbol value for this token.
+        referencePrice:
+          $ref: "#/components/schemas/NativePrice"
+        availableBalance:
+          description: |
+            The balance held by the Settlement contract that is available
+            during a settlement.
+          allOf:
+            - $ref: "#/components/schemas/TokenAmount"
+        trusted:
```

Comment:
I think `trusted` should not be optional. If we encounter an error while determining whether a token is trusted or not we should err on the safe side and flag it as untrusted.